src/DataGrid.tsx

```diff
@@ -405,10 +406,10 @@ function DataGrid<R, SR, K extends Key>(
     }
 
     prevSelectedPosition.current = selectedPosition;
-    scrollToCell(selectedPosition);
 
     if (selectedPosition.idx === -1) {
       rowRef.current!.focus({ preventScroll: true });
+      scrollIntoView(rowRef.current);
```

Comment:
question: why do we not use
`rowRef.current!.focus()` only for scroll?
I am not sure why we preventScroll in focus and scroll with another function